### PR TITLE
docs: remove http2 experimental status (#4142)

### PR DESCRIPTION
**Pull request by SebastianZimmer**
#### Checklist

- [ ] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Review by Eomm — APPROVED** [review]

File: docs/Reference/HTTP2.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 ## HTTP2
 
-_Fastify_ offers **experimental support** for HTTP2 starting from Node 8 LTS,
+_Fastify_ supports HTTP2 starting from Node 8 LTS,
 which includes HTTP2 without a flag; HTTP2 is supported over either HTTPS or
 plaintext.
```
Comment (mcollina) [inline docs/Reference/HTTP2.md:7]:
```suggestion
_Fastify_ supports HTTP2 over either HTTPS or plaintext (h2).
```

I would simplify the full sentence

Comment (SebastianZimmer) [reply docs/Reference/HTTP2.md:7]: Simplifying sounds good. But don't you mean `HTTPS (h2) or plaintext (h2c)` instead?

Comment (mcollina) [reply docs/Reference/HTTP2.md:7]: Good spot, I meant `h2c`.

**Review by mcollina — APPROVED** [review]
lgtm

**Pull request merged**